hack/e2e.sh

```diff
@@ -13,31 +13,38 @@ fi
 
 if [[ -z "$SUITE" || "$SUITE" == "enduser" ]]; then
   echo "Running end user e2e tests"
-  # Login as osadmin to simulate a regular user
+  # Login as enduser to simulate a regular user
   password=$(awk '/^  endUserPasswd:/ { print $2 }' <_data/containerservice.yaml)
   fqdn=$(awk '/^  fqdn:/ { print $2 }' <_data/containerservice.yaml)
   export KUBECONFIG=_data/_out/enduser.kubeconfig
+  # oc login is going to create the enduser.kubeconfig for us
   oc login $fqdn --username enduser --password $password --insecure-skip-tls-verify=true
-  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[EndUser\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e -kubeconfig ../../_data/_out/enduser.kubeconfig "${ARTIFACT_FLAG:-}"
+  export KUBECONFIG=../../_data/_out/enduser.kubeconfig
+  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[EndUser\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e "${ARTIFACT_FLAG:-}"
   oc logout
 fi
 
 if [[ -z "$SUITE" || "$SUITE" == "clusterreader" ]]; then
   echo "Running azure cluster reader e2e tests"
   (awk '/^  azureClusterReaderKubeconfig:/ { print $2 }' <_data/containerservice.yaml | base64 -d) > _data/_out/azure-cluster-reader.kubeconfig
-  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[AzureClusterReader\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e -kubeconfig ../../_data/_out/azure-cluster-reader.kubeconfig
+  export KUBECONFIG=../../_data/_out/azure-cluster-reader.kubeconfig
+  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[AzureClusterReader\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e
 fi
 
 if [[ -z "$SUITE" || "$SUITE" == "customer-cluster-admin" ]]; then
   echo "Running azure customer-cluster-admin tests"
   fqdn=$(awk '/^  fqdn:/ { print $2 }' <_data/containerservice.yaml)
-  export KUBECONFIG_CUSTOMER_ADMIN=_data/_out/customer-cluster-admin.kubeconfig
-  export KUBECONFIG_CUSTOMER_READER=_data/_out/customer-cluster-reader.kubeconfig
-  export KUBECONFIG_END_USER=_data/_out/enduser.kubeconfig
-
-  # oc login is going to create the kubeconfig with the proper permissions
-  KUBECONFIG=$KUBECONFIG_CUSTOMER_ADMIN oc login $fqdn --username customer-cluster-admin --password "$(awk '/^  customerAdminPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
-  KUBECONFIG=$KUBECONFIG_CUSTOMER_READER oc login $fqdn --username customer-cluster-reader --password "$(awk '/^  customerReaderPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
-  KUBECONFIG=$KUBECONFIG_END_USER oc login $fqdn --username enduser --password "$(awk '/^  endUserPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
-  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[CustomerAdmin\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e -kubeconfig ../../_data/_out/enduser.kubeconfig "${ARTIFACT_FLAG:-}"
+
+  export KUBECONFIG=_data/_out/customer-cluster-admin.kubeconfig
```

Comment:
Honestly I don't know how running these tests in production is going to look like. I like the approach of trimming our input to the tests down to a single kubeconfig but I would prefer we mimic how we are going to run these in prod and I feel we are going to end up using separate kubeconfigs. @kwoodson thoughts?

Reply:
I like the idea of a single kubeconfig but I'm not sure how it is possible.  The admin test requires separate kubeconfigs.  We specifically need to test a create from an enduser and verify the permissions are then applied correctly through the customer-admin kubeconfig.  Maybe we should have a meeting to discuss what prod tests and our e2e test strategy.

Reply:
The only reason I can think of is that these kubeconfig are going to be generated separately so in prod we will end up needing to run them separately as opposed to merging them into one.

Reply:
@kargakis makes sense.